sympy/solvers/ode.py

```diff
@@ -4256,6 +4238,16 @@ def _remove_redundant_solutions(eq, solns, order, var):
     def is_special_case_of(soln1, soln2):
         return _is_special_case_of(soln1, soln2, eq, order, var)
 
+    # We can't check the redundancy of the series type of solutions.
```

Comment:
Maybe we can check these using `getO` and `removeO`:
```julia
In [10]: e = 1 + x + O(x**2)                                                                                                      

In [11]: e.removeO()                                                                                                              
Out[11]: x + 1

In [12]: e.getO()                                                                                                                 
Out[12]: 
 ⎛ 2⎞
O⎝x ⎠
```
We can say that one solution is a special case of the other if they have the same O term and if the expression without the O is a special case of the other.

Reply:
That could be done inside `is_special_case_of`.